docs/aws/services/dynamodb-pricing.md

```diff
@@ -0,0 +1,121 @@
+
+
+## Write Type
+
+Standard writes are relatively straightforward and include single item writes (`table.put_item`) and batch writes (`batch.put_item`).
+
+Transactions (`client.execute_transaction`) group up to 25 writes (or reads, updates, or deletes) together and *guarantee that they succeed or fail together.*
+
+For a given write of an item up to *1 KB* in size:
+
+| Type                 | Cost                       |
+|----------------------|----------------------------|
+| Standard single item | 1 WRU                      |
+| Standard batch       | 1 WRU (per item)           |
+| Transactional        | 2 WRU (2x)                 |
+| Oversize 4 KB item   | 4 WRU (4x, size dependent) |
+
+
+## Read Type
+
+Part of what makes DynamoDB a compelling offering is its hybrid approach to the CAP theorem - it can adjust between eventually and strongly Consistent as needed.
+
+Wherever acceptable to the business needs and current data modeling, it is faster and cheaper to use eventually consistent reads.
+
+That said, some business logic unequivocally dictates strongly consistent reads (for example: an ATM reading a customer's balance).
+
+For a given read of an item up to *4 KB* in size:
+
+| Type                  | Cost       |
+|-----------------------|------------|
+| Eventually Consistent | 0.5 RRU    |
+| Strongly Consistent   | 1 RRU (2x) |
+| Transactional         | 2 RRU (4x) |
+
+
+## Read Operation
+
+
+Getting a single item is as simple as providing its `partition_key` (and `sort_key`, if the table has one)
+
+Queries, however, are much more involved. NoSQL databases like DynamoDB can require significant upfront data modeling work to enable the query flexibility that SQL-based databases have by default.
+
+Scans require reading the entire table, and are correspondingly slow and expensive. Wherever possible, *avoid scanning Dynamo tables.*
+
+| Type      | Cost                                 |
+|-----------|--------------------------------------|
+| `GetItem` | 1 RRU                                |
+| `Query`   | {# of items meeting query logic} RRU |
+| `Scan`    | {# of items in table} RRU            |
+
+
+## Indexes
+
+Every DynamoDB table has a `partition_key` (`pk`) and optional `sort_key` (`sk`) specified at the time of creation.
+
+Indexes allow alternate partition and sort keys to be used to query items. They may be created at any time and are automatically maintained as new items are written.
+
+Indexes can help control costs in two primary ways:
+
+1. Queries on a new index return less unnecessary items (than the alternative/existing query) and thus cost less RRUs.
+
+2. Each Index optionally allows a subset of item attributes to be **Projected** to that index. Projecting a subset can save on read costs if items are regularly >4 KB, but the projected attribute names+values sum to <4KB
```

Comment:
I believe both "Index" and "Projected" don't need to be capitalized in the first sentence here.

Reply:
Updated in the newest commit